WalletWasabi.Tests/UnitTests/BitcoinCore/P2pBasedTests.cs

```diff
@@ -137,7 +137,8 @@ public async Task BlockNotifierTestsAsync()
 		{
 			var coreNode = await TestNodeBuilder.CreateAsync();
 			using HostedServices services = new();
-			services.Register<BlockNotifier>(new BlockNotifier(TimeSpan.FromSeconds(7), coreNode.RpcClient, coreNode.P2pNode), "Block Notifier");
+			using var blockNotifier = new BlockNotifier(TimeSpan.FromSeconds(7), coreNode.RpcClient, coreNode.P2pNode);
+			services.Register<BlockNotifier>(blockNotifier, "Block Notifier");
```

Comment:
This is not necessary because of L139.

* It would be better to use:

```csharp
#pragma warning disable CA2000 // Dispose objects before losing scope
			services.Register<BlockNotifier>(new BlockNotifier(TimeSpan.FromSeconds(7), coreNode.RpcClient, coreNode.P2pNode), "Block Notifier");
#pragma warning restore CA2000 // Dispose objects before losing scope
```

as this is actually the most correct. However, it's visually ugly.

* Your solution removes the CA2000 warning and it does not introduce any issues. However, `blockNotifier` is diposed actually twice (first time the disposal happens and second time it is ignored. Not an issue per se but also slightly weird and somewhat confusing too.

So it's slightly technical and @molnard may have some opinion on the code style here.

BTW: Good read for this topic is https://github.com/dotnet/runtime/issues/29631.

Reply:
Yep. I agree.